### PR TITLE
Introduce Tenderly for Development

### PR DESCRIPTION
**Pull request by dxganta**
Tenderly has a nice visual debugger which we can use to debug local transactions too. 

- changes to the hardhat.config.js file inside docker for tenderly support
- addition of tenderly's persistArtifacts method to the scripts in deploy directory (required for tenderly to debug transactions)
- adding Tenderly installation steps to Readme.MD

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -118,25 +118,59 @@ the steps.
...
 
+### [Tenderly](https://tenderly.co/)
+
+We are also using Tenderly's Visual Debugger to debug local transactions efficiently.
+
+First you need to create a [tenderly account](https://dashboard.tenderly.co/register).
+
+Then if not installed already, [install tenderly-cli](https://github.com/Tenderly/tenderly-cli#installation) and log in.
+
+To debug any transaction, you need to be inside the docker folder first.
```
Comment (gabrielpoca) [inline README.md:154]: can we maybe have a bin/script to perform these?

File: hardhat.config.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ import "@typechain/hardhat";
 import "hardhat-deploy";
 import "hardhat-gas-reporter";
 import "solidity-coverage";
+import "@tenderly/hardhat-tenderly";
```
Comment (0xrin1) [inline hardhat.config.ts:12]: I think we can remove this from here now that we discovered that tenderly needs to use the js hardhat config that we use to spin up the hardhat node

File: README.md (hunk 1 above)
Comment (dxganta) [reply README.md:154]: @0xrin1 can answer this better

File: hardhat.config.ts (hunk 2 above)
Comment (dxganta) [reply hardhat.config.ts]: done

File: README.md (hunk 1 above)
Comment (gabrielpoca) [reply README.md:154]: ahahah I meant to say that can you please write a shell script that does these steps taking the tx hash an argument?

File: hardhat.config.ts (hunk 2 above)
Comment (0xrin1) [reply hardhat.config.ts]: yeah i think you're right, linters etc are complaining without it

File: bin/tenderly-debug [hunk 3]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash -ue
+
+cd docker
+
+file=./tenderly.yaml
+if [ ! -e "$file" ]; then
```
Comment (0xrin1) [inline bin/tenderly-debug:6]: i think you can just put ./tenderly.yaml directly in the if statement without the variable here

File: hardhat.config.ts [hunk 4]
```diff
@@ -1,22 +1,23 @@
...
-import "@typechain/hardhat";
-import "hardhat-deploy";
-import "hardhat-gas-reporter";
-import "solidity-coverage";
+import '@nomiclabs/hardhat-ethers';
+import '@nomiclabs/hardhat-waffle';
+import '@nomiclabs/hardhat-web3';
+import '@nomiclabs/hardhat-etherscan';
+import '@primitivefi/hardhat-dodoc';
+import '@typechain/hardhat';
```
Comment (gabrielpoca) [inline hardhat.config.ts:8]: can you revert these changes from double quote to single quote?

Comment (0xrin1) [reply hardhat.config.ts:8]: that was from me actually, are we enforcing double quotes on the sc repo?

Comment (gabrielpoca) [reply hardhat.config.ts:8]: apparently not, because I see the prettierrc, ignore me, move along 🚃

File: README.md (hunk 1 above)
Comment (gabrielpoca) [reply README.md]: nope, it's good!

**Review by gabrielpoca — APPROVED** [review]
